lib/util/git/index.ts

```diff
@@ -628,7 +628,10 @@ export async function isBranchBehindBase(
   }
 }
 
-export async function isBranchModified(branchName: string): Promise<boolean> {
+export async function isBranchModified(
+  branchName: string,
+  baseBranch: string | undefined,
```

Comment:
i think when `baseBranch` is undefined then we need to use `config.defaultBranch` 🤔

Reply:
Is your recommendation to use config.defaultBranch for the onboarding and migration flows and make this required rather than optional?

Reply:
Yes, I'd prefer that